### ISSUE TITLE
withSelectedItem does not behave as expected

### ISSUE THREAD

**Issue by mshi** [mikepenz/MaterialDrawer#578]
Calling with `withSelectedItem` in `DrawerBuilder` does not work as expected. It always defaults to 1.

Offending code is: 
```java
           if(this.mHeaderView != null && this.mSelectedItem != -1) {
                this.mSelectedItem = 1;
            }
``` 
in `DrawerBuilder` class.

It always sets the selected item to 1.

**Comment by mikepenz** [comment]
This method sets the selection by identifier not by position.

**Comment by mshi** [comment]
I'm referring to the builder method, not the Drawer instance method to set selection. 

From the code, it definitely looks like it is assigning it by position. However, the problem is, before the quoted portion of the code, the mSelectedItem value is correct, and after that check/assign, it goes to 1. It then calls `DrawerUtils.setRecyclerViewSelection(this, this.mSelectedItem, false);` to set the position, which here would always be 1.

**Comment by mikepenz** [comment]
Thanks. 

I'm currently not at home. Will take a look as soon as I have access to my pc today or tomorrow. 

Sorry for the issue

**Comment by mikepenz** [comment]
@mshi yes you are right. you are just not able to select anything else than the first position if you have a header set. i will adjust this to check for mSelectedItem == 0. 

will be fixed in v4.0.0
